apps/mobile/src/components/app-update-required/update-button.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import { Alert } from 'react-native';
+
+import { getStoreButtonText, handleStoreRedirect } from '@/utils/store-utils';
+import { t } from '@lingui/macro';
+
+import { Button } from '@leather.io/ui/native';
+
+interface UpdateButtonProps {
+  onPress?: () => void;
+}
+
+export function UpdateButton({ onPress }: UpdateButtonProps) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function handlePress() {
+    if (onPress) {
+      onPress();
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
```

Comment:
Would love to see a basic state machine here too instead of `try/catch` for control flow.

Reply:
I've consolidated this a ton and there is really very little state now.